### ISSUE TITLE
Apply old distortion to 2012 data before December change in CoMP

### ISSUE THREAD

**Issue by detoma** [NCAR/comp-pipeline#138]
We will first test this on a couple of days. Compare to days with no distortion correct.

Test dates:

- 2012 03 28
- 2012 06 20
- 2012 08 04
- 2012 10 16

#### Tasks

Reprocess the dates with the old distortion, results in `old-distortion`:

- [x] 20120328
- [x] 20120620
- [x] 20120804
- [x] 20121016
- [x] 20120328 with intermediate product for `fgplus` and `fgminus` in `comp_combine_beams`

Reprocess the same dates with no distortion, results in `no-distortion`:

- [x] 20120328
- [x] 20120620
- [x] 20120804
- [x] 20121016

Reprocess the same dates with new ("file") distortion, results in `new-distortion`:

- [x] 20120328
- [x] 20120620
- [x] 20120804
- [x] 20121016
- [x] 20120328 with intermediate product for `fgplus` and `fgminus` in `comp_combine_beams`

**Comment by mgalloy** [comment]
New results with intermediate products are in `engineering.{new,old}-distortion/2012/03/28`.

**Comment by jburkepile** [comment]
Dear Giuliana and Mike:

I have looked at 3 of the 4 days comparing the new and old distortions in the directories you listed below. I agree they are slightly different but I cannot tell which is 'better'. I looked at level 1 data both total intensity and Stokes Q. Giuliana has correctly pointed out that the polarization is more sensitive to proper alignment of the images. I have attached scans from Stokes Q comparing the new and old distortions for scans at the east  limb, southeast limb, west limb, north pole and south pole. They are slightly different (as expected) but I cannot say that one is clearly better than the other. One thing I will point out, there is no reason to expect that the distortion correction that starts in December should be the 'right one' for the data that preceded it. Some new optics were installed and there was some realigning to reduce scatter. It is a bit of a surprise to me that the old and new distortions give such similar results. Obviously the changes made did not dramatically change the distortion. I did notice that the very faint scattered light pattern,  best seen by looking in the region 'under' the occulter, differs slightly between the old and new distortion corrections. I noticed it when I toggled between two Stokes Q images: one with the old and one with the new distortion.
Here are the comparison scans from level 1 Stokes Q: 

![20120328_192749_stokesq_1074 62_old_minus_new_distortion](https://github.com/NCAR/comp-pipeline/assets/18175973/daeaf229-b51d-427d-a17e-f2f70820c6a7)
![20120328_192749 comp_distortion_comparison_north_pole](https://github.com/NCAR/comp-pipeline/assets/18175973/f345251f-92d3-4e20-831f-1dd2fc780685)
![20120328_192749 comp_distortion_comparison_west_limb](https://github.com/NCAR/comp-pipeline/assets/18175973/0585adc1-b9df-48de-97d1-d2bcb272c14c)
![20120328_192749 comp_distortion_comparison_south_pole](https://github.com/NCAR/comp-pipeline/assets/18175973/790daf35-bb4c-4dba-afda-da9c34e3243c)
![20120328_192749 comp_distortion_comparison_southeast-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/b43dfa9f-e596-40c1-b22a-e0bea710483e)
![20120328_192749 comp_distortion_comparison_east-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/5634768b-7576-4192-bc34-5d61c62a7baa)
![20120620_172039 comp_distortion_comparison_north-pole](https://github.com/NCAR/comp-pipeline/assets/18175973/6950973c-8b40-43a9-bcf1-235045c179f0)
![20120620_172039 comp_distortion_comparison_west-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/dedd64c8-0ae7-444b-b33f-474cc2213219)
![20120620_172039 comp_distortion_comparison_south_pole](https://github.com/NCAR/comp-pipeline/assets/18175973/5ae99451-f600-4033-acb7-5807cc5b67a6)
![20120620_172039 comp_distortion_comparison_southeast-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/92126d2d-bd1b-48d6-ab0f-eef42481210a)
![20120620_172039 comp_distortion_comparison_east-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/d2f0904f-55db-402b-913a-5b36f273fb43)
![20121016 180141 comp_distortion_comparison_north-pole](https://github.com/NCAR/comp-pipeline/assets/18175973/1962fd60-6ffc-4921-9161-ba0249bea2a7)
![20121016 180141 comp_distortion_comparison_west-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/160a1331-a53d-42ed-9725-0c6c67ca2583)
![20121016 180141 comp_distortion_comparison_south-pole](https://github.com/NCAR/comp-pipeline/assets/18175973/ec934ead-5e64-45dd-954b-c8f67adf092f)
![20121016 180141 comp_distortion_comparison_east-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/cee991d8-24b5-492e-8728-5b3304f86385)
![20121016 180141 comp_distortion_comparison_southeast-limb](https://github.com/NCAR/comp-pipeline/assets/18175973/0225fb4d-768b-444f-a968-9059d7a5a0fe)

**Comment by detoma** [comment]
Both Joan and I looked at the "old" vs "new" distortion correction and neither one is clearly better than the other. Beside intensity images, Joan looked at the stokes Q and I looked at intermediate products and created difference images of two cameras before they are combined. For both the old and new distortion the difference was non-zero but it was unclear which one was best and if the difference between the cameras was due to the distortion not being accurate or to the seeing changing in the two non-simultaneous images.

We expect that neither the old nor the new distortion will perfectly correct the data. The old distortion was based on the instrument set-up at Sac Peak and the new distortion was based on data taken after significant instrument changes made in December 2012. Unfortunately, there was no distortion calibration made at MLSO before the December 2012 changes so we do not have distortion grid data to compute the proper distortion correction for 2011-2012.

What is clear is that processing the data without distortion gives blurry images.

Mike has already reprocessed to the level 1 the 2012 COMP data with the new distortion.

Given the situation above and to save time, I think we should use for the years 2011 and 2012 the new distortion based on the grid data taken at MLSO.
